packages/@aws-cdk/aws-events/lib/connection.ts

```diff
@@ -0,0 +1,423 @@
+   */
+  readonly connectionName: string;
+
+  /**
+   * The ARN of the connection created.
+   */
+  readonly connectionArn: string;
+
+  /**
+   * The ARN for the secret created for the connection.
+   */
+  readonly connectionSecretArn: string;
+}
+
+/**
+ * Define an EventBridge Connection
+ *
+ * @resource AWS::Events::Connection
+ */
+export class Connection extends Resource implements IConnection {
+  /**
+   * Import an existing connection resource
+   * @param scope Parent construct
+   * @param id Construct ID
+   * @param connectionArn ARN of imported connection
+   */
+  public static fromEventBusArn(scope: Construct, id: string, connectionArn: string, connectionSecretArn: string): IConnection {
+    const parts = Stack.of(scope).parseArn(connectionArn);
+
+    return new ImportedConnection(scope, id, {
+      connectionArn: connectionArn,
+      connectionName: parts.resourceName || '',
+      connectionSecretArn: connectionSecretArn,
+    });
+  }
+
+  /**
+   * Import an existing connection resource
+   * @param scope Parent construct
+   * @param id Construct ID
+   * @param attrs Imported connection properties
+   */
+  public static fromConnectionAttributes(scope: Construct, id: string, attrs: ConnectionAttributes): IConnection {
+    return new ImportedConnection(scope, id, attrs);
+  }
+
+  /**
+   * The Name for the connection.
+   * @attribute
+   */
+  public readonly connectionName: string;
+
+  /**
+   * The ARN of the connection created.
+   * @attribute
+   */
+  public readonly connectionArn: string;
+
+  /**
+   * The ARN for the secret created for the connection.
+   * @attribute
+   */
+  public readonly connectionSecretArn: string;
+
+  constructor(scope: Construct, id: string, props: ConnectionProps) {
+    super(scope, id, {
+      physicalName: props.connectionName || Lazy.string({
+        produce: () => this.generateUniqueName(),
+      }),
+    });
+
+    let authParameters: any;
+
+    if (props.authorizationType === AuthorizationType.API_KEY) {
+      if (props.authParameters?.apiKeyAuthParameters !== undefined) {
+        authParameters = {
+          ApiKeyAuthParameters: {
+            ApiKeyName: props.authParameters?.apiKeyAuthParameters?.apiKeyValue,
+            ApiKeyValue: props.authParameters?.apiKeyAuthParameters?.apiKeyValue,
+          },
+        };
+      } else {
+        throw new Error('You must supply the relevant authParameters for this authorizationType');
+      };
+    };
+
+    if (props.authorizationType === AuthorizationType.BASIC) {
+      if (props.authParameters?.basicAuthParameters !== undefined) {
+        authParameters = {
+          BasicAuthParameters: {
+            Password: props.authParameters?.basicAuthParameters?.password,
+            Username: props.authParameters?.basicAuthParameters?.username,
+          },
+        };
+      } else {
+        throw new Error('You must supply the relevant authParameters for this authorizationType');
```

Comment:
You can be more specific than this! You know the auth type so you know what field they need to be passing.